### PR TITLE
Starting pass at clang-format

### PR DESCRIPTION
**Pull request by adam4813**
Run the code base through clang-format

## Description

Using a tool to automated code formatting will help make reviews easier and reduce noise.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: client/os.cpp [hunk 1]
```diff
@@ -1,539 +1,522 @@
...
-			glfwTerminate();
-			l->critical("[OS] Can't initialize window");
-			return false;
-		}
+bool OS::InitializeWindow(const int width,
+	const int height,
+	const std::string title,
+	const int glMajor /*= 3*/,
+	const int glMinor /*= 3*/,
+	bool _fullscreen /*= false*/) {
```
Comment (Meisaka) [inline client/os.cpp:37]: extra line break or something here would be nice

File: common/string.cpp [hunk 2]
```diff
@@ -14,57 +14,57 @@ namespace tec {
...
 	const char* _Source = str.c_str();
 	size_t _Dsize = std::mbstowcs(NULL, _Source, 0) + 1;
-	wchar_t *_Dest = new wchar_t[_Dsize];
+	wchar_t* _Dest = new wchar_t[_Dsize];
 	std::wmemset(_Dest, 0, _Dsize);
-	std::mbstowcs(_Dest,_Source,_Dsize);
+	std::mbstowcs(_Dest, _Source, _Dsize);
 	std::wstring result = _Dest;
-	delete []_Dest;
+	delete[] _Dest;
```
Comment (Meisaka) [inline common/string.cpp:64]: viva la clang :100:

File: client/imgui-system.cpp [hunk 3]
```diff
@@ -9,404 +9,419 @@
...
+
+	const GLchar* _fragment_shader = "#version 330\n"
+									 "uniform sampler2D Texture;\n"
+									 "in vec2 Frag_UV;\n"
+									 "in vec4 Frag_Color;\n"
+									 "out vec4 Out_Color;\n"
+									 "void main()\n"
+									 "{\n"
+									 "	Out_Color = Frag_Color * texture( Texture, Frag_UV.st);\n"
+									 "}\n";
```
Comment (Meisaka) [inline client/imgui-system.cpp:195]: I might recommend overriding this or putting it in a .inc file or something

File: client/game.cpp [hunk 4]
```diff
@@ -1,165 +1,152 @@
...
-					});
-			});
-	}
+using networking::MessageType;
+using networking::ServerMessage;
+
+Game::Game(OS& _os, std::string config_file_name) :
+	stats(), os(_os), game_state_queue(this->stats), server_connection(this->stats),
+	ps(this->simulation.GetPhysicsSystem()), vcs(this->simulation.GetVComputerSystem()),
+	sound_thread([this]() { ss.Update(); }) {
```
Comment (Meisaka) [inline client/game.cpp:25]: this could also use something to break it up a little

**Review by Meisaka — COMMENTED** [review]
rip file history

File: client/imgui-system.cpp (hunk 3 above)
Comment (adam4813) [reply client/imgui-system.cpp:195]: I think github is making this look a bit worse than it is, but you that's a good call out.

**Pull request closed without merging**